precompiles/nfts/swap/src/tests.rs

```diff
@@ -33,15 +63,94 @@ fn selectors() {
 
 #[test]
 fn create_swap_works() {
-	todo!()
+	ExtBuilder::<Runtime>::default()
+        .with_balances(vec![(ALICE.into(), 1000), (BOB.into(), 1000)])
+        .build_with_collections()
+        .execute_with(|| {
+			mint_each_collections();
+
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+			_precompiles().prepare_test(
+				ALICE,
+				Precompile1,
+				PCall::create_swap {
+					offered_collection: 0.into(),
+					offered_item: 1.into(),
+					desired_collection: 1.into(),
+					maybe_desired_item: OptionalU256::from(Some(U256::from(1))),
+					maybe_price: OptionalPriceWithDirection::from(Some(PriceWithDirection::new(U256::from(100), PriceDirection::Receive))),
+					duration: 100.into(),
+				},
+			).execute_returns(true);
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+        });
 }
 
 #[test]
 fn cancel_swap_works() {
-	todo!()
+	ExtBuilder::<Runtime>::default()
+        .with_balances(vec![(ALICE.into(), 1000), (BOB.into(), 1000)])
+        .build_with_collections()
+        .execute_with(|| {
+			mint_each_collections();
+
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+			_precompiles().prepare_test(
+				ALICE,
+				Precompile1,
+				PCall::create_swap {
+					offered_collection: 0.into(),
+					offered_item: 1.into(),
+					desired_collection: 1.into(),
+					maybe_desired_item: OptionalU256::from(Some(U256::from(1))),
+					maybe_price: OptionalPriceWithDirection::from(Some(PriceWithDirection::new(U256::from(100), PriceDirection::Receive))),
+					duration: 100.into(),
+				},
+			).execute_returns(true);
+			_precompiles().prepare_test(
+				ALICE,
+                Precompile1,
+                PCall::cancel_swap {
+                    offered_collection: 0.into(),
+                    offered_item: 1.into(),
+                },
+			).execute_returns(true);
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+        });
 }
 
 #[test]
 fn claim_swap_works() {
-	todo!()
+	ExtBuilder::<Runtime>::default()
+		.with_balances(vec![(ALICE.into(), 1000), (BOB.into(), 1000)])
+		.build_with_collections()
+		.execute_with(|| {
+			mint_each_collections();
+
+			assert_eq!(get_owner_of_item(0, 1), Some(ALICE.into()));
+			_precompiles().prepare_test(
+				ALICE,
+				Precompile1,
+				PCall::create_swap {
```

Comment:
In the context of a unit test, this test shouldn't rely on another precompile call tested here. Please instead use the `create_swap`call directly from the pallet as u can do for the mint stuff etc...

Reply:
do i need to use prepare_test or an assrt_ok! ?

Reply:
assert_ok then